File: src/Lession5/Ex9.java
package Lession5;

public class Ex9 {
    static void main() {
        int[] array = new int[]{0, 1, 2, 3, 4, 5, 6, 5, 4, 3, 2, 1, 0};

        if (checkForPalindrome(array)) {
            System.out.println("Array is a palindrone");
        } else {
            System.out.println("Array is not a palindrome");
        }
    }

    static boolean checkForPalindrome (int[] arr) {
        for (int k = 0; k < arr.length / 2; k++) {
            if (arr[k] != arr[arr.length - 1 - k]) {
                return false;
            }
        }
        return true;
    }
}

File: src/Lession5/Ex9.kt
package Lession5

fun main() {
    val array = intArrayOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 9, 8, 7, 6, 6, 4, 3, 2, 1)

    if (checkForPalindrome(array)) {
        println("Array is a palindrome")
    } else {
        println("Array is not a palindrome")
    }
}

fun checkForPalindrome (arr: IntArray) : Boolean {
    for (i in 0 until arr.size / 2) {
        if (arr[i] != arr[arr.size - 1 - i])  {
            return false
        }
    }
    return true
}